src/data_provider/src/osinfo/sysOsParsers.cpp

```diff
@@ -394,13 +390,13 @@ bool MacOsParser::parseSwVersion(const std::string& in, nlohmann::json& output)
 bool MacOsParser::parseSystemProfiler(const std::string& in, nlohmann::json& output)
 {
     constexpr auto SEPARATOR{':'};
-    static const std::map<std::string, std::string> KEY_MAP
+    static const std::vector<std::pair<std::string, std::string>> KEY_MAPPING
```

Comment:
The changes made to this function, while elementary, exceed the original scope of the issue